terminaltables.py

```diff
@@ -22,21 +22,56 @@
 import re
 import struct
 import sys
+import unicodedata
+
+PY2 = sys.version_info[0] == 2
 
 try:
     import fcntl
     import termios
-except ImportError:
+except ImportError:  # pragma: no cover
     fcntl = None
     termios = None
 
-from colorclass import _WindowsCSBI
+from colorclass import _WindowsCSBI, Color
 
 __author__ = '@Robpol86'
 __license__ = 'MIT'
 __version__ = '1.2.0'
 
 
+if PY2:  # pragma: no cover
+    def _to_unicode(string):
+        if isinstance(string, str):
+            return string.decode('u8')
+        return string
+else:  # pragma: no cover
+    _to_unicode = lambda x: x
+
+
+def _get_width(string):
+    """Get the real width of unicode string.
+
+    Positional arguments:
+    string -- unicode string.
+
+    Returns:
+    String width.
+    """
+    if isinstance(string, Color):
```

Comment:
Can you do this without isinstance? I don't like isinstance, it makes code inflexible and hard to use in other libraries. What if someone made something like colorclass where they subclass string/unicode? terminaltables wouldn't work with their library.